benchmarks/usecases.py

```diff
@@ -0,0 +1,102 @@
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Benchmarks for some use cases, typically at datalad.api level"""
+
+import sys
+import tempfile
+from datalad.utils import get_tempfile_kwargs
+import os.path as osp
+from os.path import join as opj
+
+from datalad.api import create
+
+from datalad.utils import (
+    create_tree,
+    rmtree,
+)
+
+from .common import SuprocBenchmarks
+
+
+class study_forrest(SuprocBenchmarks):
+    """
+    Benchmarks for Study Forrest use cases
+    """
+
+    timeout = 180  # especially with profiling might take longer than default 60s
+
+    def setup(self):
+        sys.stderr.write("IN SETUP!")
+        self.path = tempfile.mkdtemp(**get_tempfile_kwargs({}, prefix='bm_forrest'))
+
+    def teardown(self):
+        sys.stderr.write("IN TEARDOWN!")
+        if osp.exists(self.path):
+            sys.stderr.write("REMOVING")
+            rmtree(self.path)
+            sys.stderr.write("REMOVED")
```

Comment:
oops, forgot that I was troubleshooting those -- will first make sure that it actually works, something wasn't entirely "kosher" IIRC

Reply:
removed